DEVELOPING.md

```diff
@@ -17,7 +32,131 @@ This project:
 
 3. uses [Swift Package Manager](https://github.com/apple/swift-package-manager) to manage dependencies for development. Cocapods is supported for *consuming* the SDK.
 
-4. uses [secret-shield](https://github.com/mapbox/secret-shield) to help block secrets such as access tokens from being exposed. Install `secret-shield` by entering `npm install -g @mapbox/secret-shield`. Install the pre-commit hook by running `scripts/install-pre-commit/install-pre-commit.sh`
+4. uses [secret-shield](https://github.com/mapbox/secret-shield) to help block secrets such as access tokens from being exposed. Setup `secret-shield` by running:
+```
+npm install -g @mapbox/secret-shield
+scripts/install-pre-commit/install-pre-commit.sh
+```
+
+5. uses CircleCI and AWS Device Farm for continuous integration.
+
+### Accessing the Maps SDK's source
+
+Clone the git repository:
+
+```
+git clone git@github.com:mapbox/mapbox-maps-ios.git && cd mapbox-maps-ios
+```
+
+### Building the Maps SDK
+
+Pre-requisite: Valid `.netrc` file located on your machine at `~/.netrc`. Please visit https://docs.mapbox.com/ios/maps/overview for additional information about setting up your `.netrc` file.
+
+Open `Package.swift` and build the MapboxMaps target.
+
+### Update dependency versions
+
+You must remember to update the Package.swift file in https://github.com/mapbox/mapbox-maps-ios/blob/main/Package.swift, not the App.xcworkspace, using the following commands:
+
+```
+xed Package.swift
+//update versions in there
+swift package update
+```
+
+### Running the Debug App
+
+In order to use the debug app, open Apps/Apps.xcworkspace and build and run the `DebugApp` target.
+
+You must provide a Mapbox access token to display Mapbox-hosted maps in the `DebugApp` test application. Add your Mapbox token to the `MBXAccessToken` key in `DebugApp/DebugApp/Info.plist`. Alternatively, create a [file at `~/mapbox`](https://docs.mapbox.com/help/troubleshooting/private-access-token-android-and-ios/#ios) containing your access token.
+
+### CircleCI
+
+CircleCI's [YAML](https://en.wikipedia.org/wiki/YAML) format can be picky. You can run `make validate` to ensure you've formatted the changes to the config file correctly. This make target will install the [CircleCI command line interface](https://circleci.com/docs/2.0/local-cli/) to validate `.circleci/config.yml`, and run `circleci config validate`
```

Comment:
I think we can remove the criticism of CircleCI's YAML format 🙃 

The advice about running `make validate` is nice though.